### PR TITLE
make enable_plugin() raise an error instead of a login entry

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

I authored this logging error and didn't realise at the time it would be systematically seen in test pipelines when yt is called on the command line.
This should fix this issue.

## PR Checklist

- [x] pass `black --check yt/`
- [x] pass `isort . --check --diff`
- [x] pass `flake8 yt/`

### CODE REVIEW

**Comment by neutrinoceros** [comment]
maybe the actual problem is that `ytcfg.getboolean("yt", "loadfieldplugins")` defaults to `True` despite the face that this parameter is only kept for backwards compatibility ?

**Comment by matthewturk** [comment]
One thing we could do is have it default to something else, and if it's unset and a plugin file is found, have it report a deprecation warning.  That would catch the "default" problem.

**Comment by neutrinoceros** [comment]
Good idea, what do you think now ?

File: yt/__init__.py [hunk 1]
```diff
@@ -176,3 +176,16 @@
 from yt.units.unit_systems import UnitSystem, unit_system_registry
 
 _called_from_pytest = False
+
+
+def _check_deprecated_parameters():
+    from yt.config import ytcfg
+    from yt.funcs import issue_deprecation_warning
+
+    if ytcfg.getboolean("yt", "loadfieldplugins"):
```
Comment (munkm) [inline yt/__init__.py:185]: If this load field plugins was added since the last release I don't think we need to issue a deprecation warning.

Comment (neutrinoceros) [reply yt/__init__.py:185]: Indeed it seems to be the opposite: this is an old option that apparently fell out of use but we have a few checks still around for backward compatibility
See this comment
https://github.com/yt-project/yt/blob/7f488b615a7cf42bdabb46eeb4eb7a259665b4e4/yt/utilities/command_line.py#L48

**Comment by neutrinoceros** [comment]
@yt-fido test this please

**Comment by neutrinoceros** [comment]
> I was secretly hoping someone would fix this.

Me too actually... until I found out it was my fault 🙈

File: yt/__init__.py (hunk 1 above)
Comment (munkm) [reply yt/__init__.py:185]: ahhh ok I thought it had been more recently added!

**Pull request merged**